### DISCUSSION TITLE
Deployment to Lambda readonly docker image

### DISCUSSION THREAD

**Discussion by furyhawk** [manujosephv/pytorch_tabular#419] [Q&A] [answered]
When deployed to a readonly docker image, I encountered `OSError(30, 'Read-only file system')` when instantiating TabularModel.  I can't get detail error message from the log. Need some suggestions on which module is creating file.

There was no error when I instantiate SKLearn model in the same env for example.

**Comment by manujosephv** [comment, marked as answer]
I'm assuming the culprit is `ExperimentRunManager`[here](https://github.com/manujosephv/pytorch_tabular/blob/8fc27ee814412fbab2d606977de93dc13a625f4a/src/pytorch_tabular/tabular_model.py#L124).

This `ExperimentRunManager` creates a temp file on disk to manage experiment versions etc. I checked and it can easily be changed to any other path in code [here](https://github.com/manujosephv/pytorch_tabular/blob/8fc27ee814412fbab2d606977de93dc13a625f4a/src/pytorch_tabular/config/config.py#L697)], but sadly, as of now, it's not exposed in the api.

**Reply by manujosephv** [reply]
You'll also have ti manage where checkpoints are stored etc. and direct them to a path where you have write access
